audioipc/src/shm.rs

```diff
@@ -46,44 +45,54 @@ impl SharedMemReader {
 }
 
 pub struct SharedMemSlice {
-    view: MmapViewSync,
+    mmap: Arc<UnsafeCell<Mmap>>,
 }
 
 impl SharedMemSlice {
     pub fn from(file: &File, size: usize) -> Result<SharedMemSlice> {
-        let mmap = Mmap::open(file, Protection::Read)?;
+        let mmap = unsafe { MmapOptions::new().map(file)? };
         assert_eq!(mmap.len(), size);
-        let view = mmap.into_view_sync();
-        Ok(SharedMemSlice { view })
+        let mmap = Arc::new(UnsafeCell::new(mmap));
+        Ok(SharedMemSlice { mmap })
     }
 
     pub fn get_slice(&self, size: usize) -> Result<&[u8]> {
         if size == 0 {
             return Ok(&[]);
         }
         // TODO: Track how much is in the shm area.
-        if size <= self.view.len() {
+        if size <= self.inner().len() {
             atomic::fence(atomic::Ordering::Acquire);
-            let buf = unsafe { &self.view.as_slice()[..size] };
+            let buf = &self.inner_mut()[..size];
```

Comment:
Does `let buf = &self.inner()[..size]` work? If we can drop `mut` here then maybe we can also drop `UnsafeCell` as well?

Reply:
Good point, thanks.  Fix pushed.